src/overlays/actors/ovl_En_Nb/z_en_nb.c

```diff
@@ -31,81 +101,659 @@ const ActorInit En_Nb_InitVars = {
+            shouldChange = true;
+        }
+    } else if (animIndex != this->animIndex) {
+        shouldChange = true;
+    }
+
+    if (shouldChange) {
+        this->animIndex = animIndex;
+        didAnimationChange = SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimationInfo, animIndex);
+        this->animPlaySpeed = this->skelAnime.playSpeed;
+    }
+
+    return didAnimationChange;
+}
+
+void func_80BBFF24(EnNb* this, PlayState* play) {
+    f32 diff;
+    s32 pad;
+
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+
+    diff = this->actor.focus.pos.y - this->actor.world.pos.y;
+    this->collider.dim.height = diff;
+    CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
+}
+
+Actor* func_80BBFF90(EnNb* this, PlayState* play) {
+    Actor* actor;
+
+    if (this->schResultValue == EN_NB_SCH_2) {
+        actor = EnNb_FindActor(this, play, ACTORCAT_NPC, ACTOR_EN_AN);
+    } else {
+        actor = &GET_PLAYER(play)->actor;
+    }
+    return actor;
+}
+
+s32 func_80BBFFD4(EnNb* this, s16 arg1) {
+    s32 ret = false;
+
+    if (ActorCutscene_GetCurrentIndex() == 0x7C) {
+        ActorCutscene_Stop(0x7C);
+        ActorCutscene_SetIntentToPlay(arg1);
+    } else if (ActorCutscene_GetCanPlayNext(arg1)) {
+        ActorCutscene_StartAndSetUnkLinkFields(arg1, &this->actor);
+        ret = true;
+    } else {
+        ActorCutscene_SetIntentToPlay(arg1);
+    }
+
+    return ret;
+}
+
+s16 func_80BC0050(EnNb* this, s32 arg1) {
+    s16 cutscene = this->actor.cutscene;
+    s32 i;
+
+    for (i = 0; i < arg1; i++) {
+        cutscene = ActorCutscene_GetAdditionalCutscene(cutscene);
+    }
+
+    return cutscene;
+}
+
+s32 func_80BC00AC(Actor* thisx, PlayState* play) {
+    EnNb* this = THIS;
+    s16 cutscene = func_80BC0050(this, 0);
+    s32 ret = 0;
+
+    switch (this->unk_288) {
+        case 0:
+            if (func_80BBFFD4(this, cutscene)) {
+                case 2:
+                case 4:
+                case 6:
+                case 8:
+                    Camera_SetTargetActor(Play_GetCamera(play, ActorCutscene_GetCurrentSubCamId(cutscene)),
+                                          &this->actor);
+                    this->unk_288++;
+                    ret = 1;
+            }
+            break;
+
+        case 1:
+        case 3:
+        case 5:
+        case 7:
+            if ((this->actor.child != NULL) && (this->actor.child->update != NULL)) {
+                Camera_SetTargetActor(Play_GetCamera(play, ActorCutscene_GetCurrentSubCamId(cutscene)),
+                                      this->actor.child);
+            }
+            this->unk_288++;
+            ret = 1;
+            break;
+
+        case 9:
+            ActorCutscene_Stop(cutscene);
+            this->unk_288++;
+            ret = 1;
+            break;
+    }
+
+    return ret;
+}
+
+s32 func_80BC01DC(Actor* thisx, PlayState* play) {
+    EnNb* this = THIS;
+    s32 pad;
+    s32 sp2C = 0;
```

Comment:
```suggestion
    s32 ret = false;
```
Also make this boolean throughout